### PR TITLE
Migrate Assembly Loader Tests and Asynchrony Primitives Tests from VSO.

### PR DESCRIPTION
**Pull request by NingnaD**

### CODE REVIEW

**Comment by jthelin** [comment]
This feels like it is 2 unrelated changes? 
Can we split into separate PRs?

File: src/TestInternalGrainInterfaces/TestInternalGrainInterfaces.csproj [hunk 1]
```diff
@@ -39,6 +39,14 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="ActivationGCTestGrainInterfaces.cs" />
```
Comment (sergeybykov) [inline src/TestInternalGrainInterfaces/TestInternalGrainInterfaces.csproj]: Looks like most of these interfaces aren't related to the test we are migrating. They are used by other tests. So let's add them in a PR with those tests.

Also, looks like most if not all of them don't use internal types and should be moved to TestGrainInterfaces instead.

File: src/TestInternalGrains/ActivateDeactivateTestGrain.cs [hunk 2]
```diff
@@ -23,13 +46,13 @@ internal class SimpleActivateDeactivateTestGrain : Grain, ISimpleActivateDeactiv
 
         public override async Task OnActivateAsync()
         {
-            this.logger = GetLogger();
+            logger = GetLogger();
```
Comment (jdom) [inline src/TestInternalGrains/ActivateDeactivateTestGrain.cs]: why all of these unnecessary changes? They make the diff harder. And personally, if anything, I would go the other way and add `this.` when it's not there. But that's a war I don't want to get into at all, as there can be 1 distinct opinion per contributor, but what I would say is let's not change code style for what it's already there even when the standard project style has not been set. Makes sense?

Comment (jthelin) [inline src/TestInternalGrains/ActivateDeactivateTestGrain.cs]: FWIW, the DotNet OSS Coding Style Guide says "avoid  [using] `this.` unless absolutely necessary." (item 4)  No war left to fight on that one! ;)
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

But, using `base.` on the other hand is still an area for individual artistic expression .....  :)

However, there are many dark recesses in the Orleans code that pre-date that style guide and haven't been altered in (literally) years. :(

It's a tough call whether bringing legacy code into compliance with the current coding style is best done in one big bang, or incrementally as people touch old code. 
In the past we have tended to do the latter (incremental fix), but if it makes a particular code review too hard then reviewers have also pushed back at times and originator will split.

Sadly the .NET Codeformatter Tool wasn't particularly useful last time we tried it out [early 2015 IIRC].

**Pull request merged**